src/nurse/views.py

```diff
@@ -57,6 +58,22 @@ class NurseViewSet(viewsets.ModelViewSet):
     serializer_class = NurseSerializer
 
 
+class UserOneSignalProfileViewSet(
+    mixins.CreateModelMixin, viewsets.ReadOnlyModelViewSet
+):
+    queryset = UserOneSignalProfile.objects.all()
+    serializer_class = UserOneSignalProfileSerializer
+    permission_classes = [IsAuthenticated]
+
+    def get_queryset(self):
+        user = self.request.user
+        return UserOneSignalProfile.objects.filter(user=user)
+
+    def create(self, request, *args, **kwargs):
+        request.data["user"] = request.user.id
+        return super().create(request, *args, **kwargs)
```

Comment:
I don't get the `request.data["user"] = request.user.id` part, what are you trying to achieve?
We need to create the one signal profile based on the request.user